src/policy/policy.cpp

```diff
@@ -208,11 +208,45 @@ bool AreInputsStandard(const CTransaction& tx, const CCoinsViewCache& mapInputs)
     return true;
 }
 
+bool IsAnnexStandard(const std::vector<unsigned char>& annex)
+{
+    // Get the size of the annex excluding the tag byte.
+    size_t annex_size = annex.size() - 1;
+
+    // Allow an empty annex. This allows inputs to opt-in to annex
+    // usage with the minimal number of bytes.
+    if (annex_size == 0) {
+        return true;
+    }
+
+    // Annexes are nonstandard as long as they don't start with a
+    // 0x00 byte which signals unstructured data.
+    if (annex[1] != 0) {
+        return false;
+    }
+
+    // Calculate the size of the unstructured data by subtracting
+    // one for the 0x00 signaling byte.
+    size_t unstructured_annex_size = annex_size - 1;
```

Comment:
My thought was that it might be good to choose a number of actually useable bytes that is a multiple of a commonly used data structure sizes such as 32 bytes. Therefore 256. It is an arbitrary number though that it supposed to strike a balance between the limiting the potency of annex inflation attacks and having enough space to make the annex useful. 

Are you proposing to limit the annex size including any tags to 257 bytes? I see the conceptual difference, but there wouldn't be a functional difference for now?

Reply:
With the constant defined as suggested below, this suggestion does seem to make reading easier. Changed.

Reply:
> Are you proposing to limit the annex size including any tags to 257 bytes? I see the conceptual difference, but there wouldn't be a functional difference for now?

Yes, for now it doesn’t make a functional difference. The issue I have in mind is the following, assuming the unstructured annex format is deployed and we see applications building on top of it, the 256 (or 257 now) annex size limit will have to become a max-size data payload, which is going to leak in the all application toolchains (in the same sense than TCP’s 64k bytes max frame is often used as the limit for application transport design).

As of today, the signaling byte size is of 1 byte, however if in the future we have a consensus definition of the tag being  _more_ than 1 byte, we might have to swallow the 255 bytes reserved for the application data-payload. Not swallowing on the 255 bytes would mean the consensus-aware application will have to pay the witness cost for the 0x00 signaling byte and this byte will have to be discarded by parsing logic.

So if those observations are correct, I think the constant could have some safety margin padding for changes of the “structured/unstructured guard” signaling byte size. One can think 256 byte reserved for application-data + 32 byte of padding, from which this 1st byte of padding will be the current `0x00` one ? And I think that way we can guarantee to the application a multiple of a 32-byte based data structure.

Reply:
I am not sure if I fully get what you mean. Can you give a concrete example of a future extended format that might interfere with the 0x00 + unstructured proposal? My thought was to simply reserve all non-0x00 first bytes for future use.

When adding 32 bytes of padding, do you mean that initially (in a world with only unstructured annex data), 31 of those bytes are wasted but paid for?

Reply:
> Can you give a concrete example of a future extended format that might interfere with the 0x00 + unstructured proposal? 

Sure any future serialization format where the semantic “tag” is encoded on more than 8-bit to allow more than 256 types of payload without having to picked up which are the “most-used” 256 one when the consensus change. This is actually the serialization format on which the TLV proposal of the annex (with some other discussed tweaks for the length), see comment above `VarIntMode` in `src/serialize.h`.

From my understanding, if we’re following your approach where non-`0x00` first bytes are reserved for future use, and let’s say `0x01` is for TLV semantic of the annex field, it means _all_ the annex payload with consensus enforcement will have to pay the feerate cost of the 1 byte marker (either `0x00` or `0x01`).

So I’m thinking the advantage of multi-bytes padding we reserve a quantitatively significant number of bytes for the eventual adoption of a multi-bytes serialization format. This comes at the burden of unstructured data having to pay for the cost of the _reserve byte padding_ *toda* while making *tomorrow* potential consensus enforced annex payload cheaper, I think.

Of course 32-byte might be considered to high a _reserve byte padding_ to make the usage of the annex attractive for unstructured data applications. Yet the question we might have to answer is how much consensus “tag” we might need to preserve for now, let’s say in the future we have an automated consensus mechanism where people can “burn” satoshis / coins to assign an identifier to an annex tag (weird use-cases that has been proposed in the early days of Bitcoin for namespaces experiment iirc). I don’t know the answer to this “how much tag" question.